### ISSUE TITLE
chi square tests for mutation verification broken

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#2316]
It looks like scipy changed the semantics of the chi square tests we that we use for mutation verification a few years ago, and now the tests don't work:
```

jk@empire$ python3 verification.py  -t 1 --test-class SeqGenTest 
  0%|                                                                                           | 0/6 [00:00<?, ?it/s]2024-08-07 16:24:12,359 [50138] INFO     root: running 6 tests using 1 processes
2024-08-07 16:24:12,359 [50138] INFO     root: Running TestInstance(test_class='SeqGenTest', method_name='test_BLOSUM62')
2024-08-07 16:24:12,486 [50138] CRITICAL root: Traceback (most recent call last):
  File "/home/jk/work/github/msprime/verification.py", line 6413, in <module>
    main()
  File "/home/jk/work/github/msprime/verification.py", line 6409, in main
    run_tests(suite, args)
  File "/home/jk/work/github/msprime/verification.py", line 6348, in run_tests
    runner.run(tests, args.output_dir, args.num_threads, not args.no_progress)
  File "/home/jk/work/github/msprime/verification.py", line 6317, in run
    self.__run_sequential(tests, basedir, progress)
  File "/home/jk/work/github/msprime/verification.py", line 6289, in __run_sequential
    test.run(basedir)
  File "/home/jk/work/github/msprime/verification.py", line 6250, in run
    method()
  File "/home/jk/work/github/msprime/verification.py", line 5663, in test_BLOSUM62
    self._run_seq_gen_msprime_comparison("BLOSUM62")
  File "/home/jk/work/github/msprime/verification.py", line 5640, in _run_seq_gen_msprime_comparison
    df_sg, df_ts, alleles = self._run_seq_gen_msprime_stats(
  File "/home/jk/work/github/msprime/verification.py", line 5613, in _run_seq_gen_msprime_stats
    tm_chisq_sg = self._transition_matrix_chi_sq(
  File "/home/jk/work/github/msprime/verification.py", line 5349, in _transition_matrix_chi_sq
    chisq = scipy.stats.chisquare(row[not_zeros], p[not_zeros])
  File "/home/jk/.local/lib/python3.10/site-packages/scipy/stats/_stats_py.py", line 7553, in chisquare
    return power_divergence(f_obs, f_exp=f_exp, ddof=ddof, axis=axis,
  File "/home/jk/.local/lib/python3.10/site-packages/scipy/stats/_stats_py.py", line 7394, in power_divergence
    raise ValueError(msg)
ValueError: For each axis slice, the sum of the observed frequencies must agree with the sum of the expected frequencies to a relative tolerance of 1e-08, but the percent differences are:
3.872673086859892e-06

```

It is probably fairly easy to fix, but I'm not sure how the values should be normalised. 

@petrelharp, do you know what's going on here?

https://github.com/scipy/scipy/issues/14298

**Comment by petrelharp** [comment]
tl;dr our use of the chisquared was fine; it looks like scipy was forging ahead without consulting any statisticians. But all we were using it for was computing the statistic anyhow; we were following R in getting a p-value by simulation, not asymptotics.
